drivers/pwm/pwm_pch_blinky.c

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2023 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT intel_blinky_pwm
+
+#include <errno.h>
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+#include <zephyr/init.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/pwm.h>
+
+#define PWM_CTRL_REG            0x304
+#define PWM_ENABLE              0x80000000
+#define PWM_SWUP                0x40000000
+#define PWM_FREQ_INT_SHIFT      22
+#define PWM_FREQ_MAX            0x100
+#define PWM_DUTY_MAX            0x100
+
+struct bk_intel_config {
+	DEVICE_MMIO_NAMED_ROM(reg_base);
+	uint32_t clock_freq;
+	uint32_t max_pins;
+};
+
+struct bk_intel_runtime {
+	DEVICE_MMIO_NAMED_RAM(reg_base);
+	uint32_t pwm_ctrl;
+};
+
+static int bk_intel_set_cycles(const struct device *dev, uint32_t pin,
+			       uint32_t period_cycles, uint32_t pulse_cycles,
+			       pwm_flags_t flags)
+{
+	uint32_t ret = 0;
+	uint32_t val;
+	uint32_t period;
+	uint32_t duty;
+	struct bk_intel_runtime *rt = dev->data;
+	const struct bk_intel_config *cfg = dev->config;
+
+	if (pin > cfg->max_pins) {
```

Comment:
if `max_pins` is the number of pins and pin index begins with 0, it should be `>=` or do pin indices begin with 1?